### PR TITLE
refactor(datepicker): move 'outsideDays' generation to the service

### PR DESCRIPTION
**Pull request by maxokorokov**
The last in series of moving logic from components to the service and utility functions

### CODE REVIEW

File: src/datepicker/datepicker-month-view.ts [hunk 1]
```diff
@@ -48,14 +48,14 @@ import {DayTemplateContext} from './datepicker-day-template-context';
       </div>
     </div>
     <ng-template ngFor let-week [ngForOf]="month.weeks">
-      <div *ngIf="!isCollapsed(week)" class="ngb-dp-week" role="row">
```
Comment (benouat) [inline src/datepicker/datepicker-month-view.ts]: Why did you remove the `role="row"` here ?

Comment (maxokorokov) [reply src/datepicker/datepicker-month-view.ts]: Unfortunate rebase, thanks for the catch!
